src/edmd/regression.jl

```diff
@@ -1,38 +1,59 @@
-function linear_regression(Y::Vector{Float64}, X::Matrix{Float64}; gamma::Float64=0.0, 
-                           lambda::Float64=0.0)
+function linear_regression(Y::AbstractVector{<:AbstractFloat}, 
```

Comment:
Added new methods for solving the linear regression problem. The cholesky one is the fastest but QR is more robust.